kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/MetricsReportSuite.scala

```diff
@@ -0,0 +1,379 @@
+
+    // Verify contents
+    assert(snapshotReport.tablePath == resolvePath(path))
+    assert(snapshotReport.operationType == "Snapshot")
+    exception match {
+      case Some(e) =>
+        assert(snapshotReport.exception().isPresent &&
+          Objects.equals(snapshotReport.exception().get(), e))
+      case None => assert(!snapshotReport.exception().isPresent)
+    }
+    assert(snapshotReport.reportUUID != null)
+    assert(Objects.equals(snapshotReport.version, expectedVersion),
+      s"Expected version $expectedVersion found ${snapshotReport.version}")
+    assert(Objects.equals(snapshotReport.providedTimestamp, expectedProvidedTimestamp))
+
+    // Since we cannot know the actual durations of these we sanity check that they are > 0 and
+    // less than the total operation duration whenever they are expected to be non-zero/non-empty
+    if (expectNonEmptyTimestampToVersionResolutionDuration) {
+      assert(snapshotReport.snapshotMetrics.timestampToVersionResolutionDuration.isPresent)
+      assert(snapshotReport.snapshotMetrics.timestampToVersionResolutionDuration.get > 0)
+      assert(snapshotReport.snapshotMetrics.timestampToVersionResolutionDuration.get <
+        duration)
+    } else {
+      assert(!snapshotReport.snapshotMetrics.timestampToVersionResolutionDuration.isPresent)
+    }
+    if (expectNonZeroLoadProtocolAndMetadataDuration) {
+      assert(snapshotReport.snapshotMetrics.loadInitialDeltaActionsDuration > 0)
+      assert(snapshotReport.snapshotMetrics.loadInitialDeltaActionsDuration < duration)
+    } else {
+      assert(snapshotReport.snapshotMetrics.loadInitialDeltaActionsDuration == 0)
+    }
+  }
+
+  test("SnapshotReport valid queries") {
+    withTempDir { tempDir =>
+      val path = tempDir.getCanonicalPath
+
+      // Set up delta table with version 0, 1
+      spark.range(10).write.format("delta").mode("append").save(path)
+      val version0timestamp = System.currentTimeMillis
+      // Since filesystem modification time might be truncated to the second, we sleep to make sure
+      // the next commit is after this timestamp
+      Thread.sleep(1000)
+      spark.range(10).write.format("delta").mode("append").save(path)
+
+      // Test getLatestSnapshot
+      checkSnapshotReport(
+        (table, engine) => table.getLatestSnapshot(engine),
+        path,
+        expectException = false,
+        expectedVersion = Optional.of(1),
+        expectedProvidedTimestamp = Optional.empty(), // No time travel
+        expectNonEmptyTimestampToVersionResolutionDuration = false, // No time travel
+        expectNonZeroLoadProtocolAndMetadataDuration = true
+      )
+
+      // Test getSnapshotAsOfVersion
+      checkSnapshotReport(
+        (table, engine) => table.getSnapshotAsOfVersion(engine, 0),
+        path,
+        expectException = false,
+        expectedVersion = Optional.of(0),
+        expectedProvidedTimestamp = Optional.empty(), // No time travel
```

Comment:
Didn't we time travel to version 0? Would `// No time travel by timestamp` be a more accurate comment? And I think this would apply to all the other such instances of this comment, right?